### ISSUE TITLE
Add support for Makefiles

### ISSUE THREAD

**Issue by kiryph** [p00f/nvim-ts-rainbow#136]
```make
docs := $(addprefix $(OUTPUT_DIR), $(notdir $(graphs:%.graphml=%.$(FORMAT))))

clean:
	-$(RM) $(foreach ext, $(CLEAN_EXTS), $(docs:%.$(FORMAT)=%.$(ext)))
```

When using functions in `Makefiles`, you can end up easily with many closing parentheses, as seen above.
Note, instead of parentheses also curly brackets can be used:

> Call functions with `$(fn, arguments)` or `${fn, arguments}`

Also support for Guile (Scheme/Lisp) has been added in GNU Make version 4.0 (2013) which adds additional constructs with parentheses

```
$(if $(guile (access? "myfile" R_OK)),$(info myfile exists))
```
https://www.gnu.org/software/make/manual/html_node/Guile-Types.html
https://www.gnu.org/software/make/manual/html_node/Guile-Example.html

Can nvim-ts-rainbow support `Makefiles`?

**Comment by p00f** [comment]
![image](https://user-images.githubusercontent.com/36493671/187990828-2466bd68-ff71-4091-bb17-9abfd45ded9e.png)

It's inconsistent - `$` in command substitution gets highlighted and I can't fix that because `command_substitution` is an invalid node even though playground calls it exactly that
![image](https://user-images.githubusercontent.com/36493671/187991375-ace6c936-c0cd-4e99-8d4e-b8327022467b.png)

**Comment by kiryph** [comment]
Should I report this issue to the treesitter grammar for makefiles?
https://github.com/alemuller/tree-sitter-make

**Comment by p00f** [comment]
Yes, go ahead, I think the author uses neovim

**Comment by kiryph** [comment]
Are we talking about the same grammar?
`:TSInstall make` and `:TSPlaygroundToggle` says for me `substitution_reference`:
```
rule [25, 0] - [28, 0]
  targets [25, 0] - [25, 5]
    word [25, 0] - [25, 5]
  recipe [25, 6] - [28, 0]
    recipe_line [26, 1] - [26, 67]
      shell_text [26, 2] - [26, 67]
        variable_reference [26, 2] - [26, 7]
          word [26, 4] - [26, 6]
        function_call [26, 8] - [26, 67]
          arguments [26, 18] - [26, 66]
            argument: text [26, 18] - [26, 66]
              variable_reference [26, 23] - [26, 36]
                word [26, 25] - [26, 35]
              substitution_reference [26, 38] - [26, 66]
                text: word [26, 40] - [26, 44]
                pattern: concatenation [26, 45] - [26, 56]
                  word [26, 45] - [26, 47]
                  variable_reference [26, 47] - [26, 56]
                    word [26, 49] - [26, 55]
                replacement: concatenation [26, 57] - [26, 65]
                  word [26, 57] - [26, 59]
                  variable_reference [26, 59] - [26, 65]
                    word [26, 61] - [26, 64]
```


I have searched the source code of https://github.com/alemuller/tree-sitter-make after the word `command_substitution` with no luck.

**Comment by p00f** [comment]
Right, `command_substitution` is in `shell_text` and is bash:

https://github.com/nvim-treesitter/nvim-treesitter/blob/73cd1f189bf19e99af9b1349cdca590a40b985b0/queries/make/injections.scm#L3

**Comment by p00f** [comment]
I'll push this then

**Comment by p00f** [comment]
Fixed by c641e224731180371e6a4705762af0c6a882d95e

**Comment by kiryph** [comment]
Thanks. Yes it works for me.
